packages/client/hmi-client/src/utils/model-configuration.ts

```diff
@@ -0,0 +1,59 @@
+import { ModelConfiguration } from '@/types/Types';
+import { cloneDeep, isEqual } from 'lodash';
+
+export const areModelConfigValuesEqual = (
+	originalConfig: ModelConfiguration | null,
+	newConfig: ModelConfiguration | null
+) => {
```

Comment:
Thanks for creating a util function. I think maybe we can just move this to `ops/model-config/model-config-operation.ts`. It is a bit more thematic and easier to find.


Secondly, you are checking if "every-item-are-the-same" is false, another way to think about it is whether "some-items-are-different" is true. If we believe that people are going to tweak value fields more than they are going to tweak metadata fields for model-configuration, then the latter option would be a more efficient check as it would short circuit exit much earlier on average.

Reply:
Sure, I've moved the function and now I'm using `.some` instead.

I've also created a function to check if the model configurations are equal, the way we were doing it originally was not working as intended.  This has been fixed with that new function `isModelConfigsEqual`.  Let me know what you think 😄